src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2041,44 +2041,83 @@ private BoundExpression BindArgumentValue(DiagnosticBag diagnostics, ArgumentSyn
             }
 
             var declarationExpression = (DeclarationExpressionSyntax)argumentSyntax.Expression;
-            var declaration = (TypedVariableComponentSyntax)declarationExpression.VariableComponent;
-            var typeSyntax = declaration.Type;
+            var typeSyntax = declarationExpression.Type();
```

Comment:
Can you please move this line and the rest of the method into an appropriately named helper? It is confusing to have `BindArgumentValue` mostly concerned with handling this one argument form.